### PR TITLE
Refactor layout

### PR DESCRIPTION
**Pull request by zamotany**

### CODE REVIEW

File: __setup__/enzyme.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable import/no-extraneous-dependencies */
 
-import Enzyme from 'enzyme';
```
Comment (satya164) [inline __setup__/enzyme.js]: So we don't use enzyme anymore? Do you plan to remove the file?

File: example/src/index.js [hunk 2]
```diff
@@ -25,4 +25,6 @@ render(<App />, process.stdout, {
   renderOptimizations: false,
   hideCursor: true,
   exitOnError: true,
+  clearOnExit: false,
```
Comment (satya164) [inline example/src/index.js]: Maybe should be `clearScreenOnExit`?

File: example/src/index.js [hunk 3]
```diff
@@ -25,4 +25,6 @@ render(<App />, process.stdout, {
...
+  clearScrollBackOnExit: true,
```
Comment (satya164) [inline example/src/index.js]: `scrollback` is a single word, so should be `clearScrollbackOnExit`

File: src/effects/clearCallbacksOnExit.js [hunk 4]
```diff
@@ -16,6 +16,13 @@ export default function clearCallbacksOnExit() {
       });
     };
 
+    Object.defineProperty(global[`set${type}`], 'name', {
+      configurable: true,
+      enumerable: true,
+      writable: true,
+      value: 'setTimeoutCustom',
```
Comment (satya164) [inline src/effects/clearCallbacksOnExit.js]: What does this do?

File: src/nodes/ContainerNode.js [hunk 5]
```diff
@@ -43,13 +40,16 @@ export default class ContainerNode {
     };
     this.stream = stream;
 
+    this.canvasSize = {
```
Comment (satya164) [inline src/nodes/ContainerNode.js]: This can change, no?

File: src/utils/sliceAnsi.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
+import isFullwidthCodePoint from 'is-fullwidth-code-point';
```
Comment (satya164) [inline src/utils/sliceAnsi.js]: Maybe some comments in this file. No idea what's happening

Comment (zamotany) [reply src/utils/sliceAnsi.js]: Neither do I, this was taken from npm package, but was broken with 256bit colors so this is just a temporary solution.

File: src/nodes/ContainerNode.js (hunk 5 above)
Comment (zamotany) [reply src/nodes/ContainerNode.js]: Yes, but I don't care about it right now. We can handle it later.

File: src/effects/clearCallbacksOnExit.js (hunk 4 above)
Comment (zamotany) [reply src/effects/clearCallbacksOnExit.js]: I thinks it's a leftover.

File: __setup__/enzyme.js (hunk 1 above)
Comment (zamotany) [reply __setup__/enzyme.js]: Yeah, I'll remove it.

**Pull request merged**